src/client.rs

```diff
@@ -777,30 +758,33 @@ impl Spotify {
     ///
     /// Parameters:
     /// - playlist_id - the id of the playlist
+    /// - uris - a list of Spotify URIs to replace or clear
     /// - range_start - the position of the first track to be reordered
+    /// - insert_before - the position where the tracks should be inserted
     /// - range_length - optional the number of tracks to be reordered (default:
     ///   1)
-    /// - insert_before - the position where the tracks should be inserted
     /// - snapshot_id - optional playlist's snapshot ID
     ///
     /// [Reference](https://developer.spotify.com/documentation/web-api/reference/#endpoint-reorder-or-replace-playlists-tracks)
     #[maybe_async]
-    pub async fn playlist_reorder_tracks<R: Into<Option<u32>>>(
+    pub async fn playlist_reorder_tracks(
         &self,
         playlist_id: &PlaylistId,
-        range_start: i32,
-        range_length: R,
-        insert_before: i32,
-        snapshot_id: Option<String>,
+        uris: Option<&[&Id<impl PlayableIdType>]>,
+        range_start: Option<i32>,
+        insert_before: Option<i32>,
+        range_length: Option<u32>,
+        snapshot_id: Option<&str>,
     ) -> ClientResult<PlaylistResult> {
-        let mut params = json! ({
-            "range_start": range_start,
-            "range_length": range_length.into().unwrap_or(1),
-            "insert_before": insert_before
-        });
-        if let Some(snapshot_id) = snapshot_id {
-            json_insert!(params, "snapshot_id", snapshot_id);
-        }
+        let uris = uris.map(|u| u.iter().map(|id| id.uri()).collect::<Vec<_>>());
+        let params = build_json! {
+            "playlist_id": playlist_id,
```

Comment:
Is it a mistake that the `playlist_id` is a part of the JSON body parameter?

Reply:
yup, removed